src/prompt/functions/ai-runner.ts

```diff
@@ -100,9 +104,11 @@ export function createAIRunner<Content extends any = string>(args: {
 
         // Check if the last message should break the loop
         const lastMessage = messages[messages.length - 1];
-        if (shouldBreakLoop(lastMessage)) {
+        if (Msg.isRefusal(lastMessage)) {
+          throw new AbortError(lastMessage.refusal);
+        } else if (shouldBreakLoop(lastMessage)) {
           const content = await Promise.resolve(
-            validateContent(lastMessage.content)
+            validateContent(lastMessage.content!)
```

Comment:
the pattern:

`if (Msg.isRefusal(message)) { ... } else { const content = message.content! }`

is unfortunately going to be a common one going forwards I think

Reply:
Shouldn't `Msg.isRefusal()` narrow the type to the point where TypeScript knows `message.content: string`?

Reply:
the else block here doesn't guarantee that `content` exists